src/Populator/ArrayPropertyMappingPopulator.php

```diff
@@ -45,12 +45,17 @@ public function __construct(
     public function populate(object $target, object $source, ?object $ctx = null): void
     {
         try {
-            $unwrappedArray = array_map(
-                fn ($item) => null !== $this->sourceArrayItemProperty
-                    ? $this->arrayItemAccessor->getValue($item, $this->sourceArrayItemProperty)
-                    : $item,
-                $this->accessor->getValue($source, $this->sourceArrayProperty),
-            );
+            $sourceArrayPropertyValues = $this->accessor->getValue($source, $this->sourceArrayProperty);
+
+            $unwrappedArray = [];
+            if (!empty($sourceArrayPropertyValues)) {
```

Comment:
Should we check the type here? `$sourceArrayPropertyValues` actually *must* be an array to work with `array_map()`.

Like:
```suggestion
            if (is_array($sourceArrayPropertyValues) && [] !== $sourceArrayPropertyValues) {
```